sdk/core/azure-core/inc/azure/core/http/policy.hpp

```diff
@@ -110,14 +138,14 @@ namespace Azure { namespace Core { namespace Http {
      * @param transport A pointer to the #HttpTransport implementation to use when this policy gets
      * applied (#Send).
      */
-    explicit TransportPolicy(std::shared_ptr<HttpTransport> transport)
-        : m_transport(std::move(transport))
+    explicit TransportPolicy(TransportPolicyOptions options = TransportPolicyOptions())
+        : m_options(std::move(options))
     {
     }
 
     std::unique_ptr<HttpPolicy> Clone() const override
     {
-      return std::make_unique<TransportPolicy>(m_transport);
+      return std::make_unique<TransportPolicy>(*this);
```

Comment:
Will this clone all the options too?

Reply:
No.
And I think this is a bug. I will create an issue to follow up

Reply:
Yes.
We lose the type, but keep a copy.
https://godbolt.org/z/Y7nGjW